### PR TITLE
feat: stabilize `Deno.createHttpClient()`

### LINKED ISSUE #25518: Consider stabilizing `Deno.createHttpClient()`
This would allow us to deprecate `--unstable-http` flag.

This API is used to create a custom HTTP client that can then be passed to `fetch` API:

```
const caCert = await Deno.readTextFile("./ca.pem");
const client = Deno.createHttpClient({ caCerts: [ caCert ] });
const req = await fetch("https://myserver.com", { client });
```

### PR DESCRIPTION
**Pull request by iuioiua**
Closes #25518

### CODE REVIEW

**Comment by lucacasonato** [comment]
There is still the open question of whether we want to change the api to `client.fetch` instead of `fetch(..., { client })`

File: cli/tsc/99_main_compiler.js [hunk 1]
```diff
@@ -44,7 +44,6 @@ delete Object.prototype.__proto__;
     "UnsafeFnPointer",
     "UnixConnectOptions",
     "UnixListenOptions",
-    "createHttpClient",
```
Comment (bartlomieju) [inline cli/tsc/99_main_compiler.js:47]: Can you remove `CreateHttpClientOptions` and `HttpClient` as well?

File: cli/tsc/dts/lib.deno.ns.d.ts [hunk 2]
```diff
@@ -6087,4 +6087,126 @@ declare namespace Deno {
...
+   * The options used when creating a {@linkcode Deno.HttpClient}.
+   *
+   * @category Fetch
+   */
+  export interface CreateHttpClientOptions {
+    /** A list of root certificates that will be used in addition to the
+     * default root certificates to verify the peer's certificate.
+     *
+     * Must be in PEM format. */
+    caCerts?: string[];
```
Comment (bartlomieju) [inline cli/tsc/dts/lib.deno.ns.d.ts:6118]: @lucacasonato suggested to align this one with other TLS APIs

Comment (bartlomieju) [reply cli/tsc/dts/lib.deno.ns.d.ts:6118]: Er I guess it's already matching the `ConnectTlsOptions`.

File: runtime/lib.rs [hunk 3]
```diff
@@ -80,56 +80,50 @@ pub static UNSTABLE_GRANULAR_FLAGS: &[UnstableGranularFlag] = &[
     show_in_help: false,
     id: 4,
   },
-  UnstableGranularFlag {
-    name: ops::http::UNSTABLE_FEATURE_NAME,
-    help_text: "Enable unstable HTTP APIs",
-    show_in_help: true,
-    id: 5,
-  },
```
Comment (bartlomieju) [inline runtime/lib.rs:88]: Please keep it, just mark `show_in_help` as `false`

**Review by bartlomieju — APPROVED** [review]
LGTM, please remove `(fetch)` scope from the commit title before landing

**Pull request merged**